From ca175ef5bf7f96db57fb5aa562602dc424fab81c Mon Sep 17 00:00:00 2001
From: Joel Winarske <joel.winarsk@gmail.com>
Date: Sun, 5 Mar 2023 10:43:02 -0800
Subject: [PATCH] fix toolchain paths

Flutter includes a copy of clang which is required to build. However, by
default, build/toolchain/custom/BUILD.gn attempts to use several gcc utilities
such as ar, readelf, nm, and strip. Change these to the proper llvm provided
utility.

Upstream: https://github.com/flutter/flutter/issues/123073

Signed-off-by: Joel Winarske <joel.winarsk@gmail.com>
Signed-off-by: Adam Duskett <adam.duskett@amarulasolutions.com>
---
 build/toolchain/custom/BUILD.gn | 4 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/build/toolchain/custom/BUILD.gn b/build/toolchain/custom/BUILD.gn
index 65b1623..8b3f6f8 100644
--- a/build/toolchain/custom/BUILD.gn
+++ b/build/toolchain/custom/BUILD.gn
@@ -12,11 +12,11 @@ toolchain("custom") {
   # these values in our scope.
   cc = "${toolchain_bin}/clang"
   cxx = "${toolchain_bin}/clang++"
-  ar = "${toolchain_bin}/${custom_target_triple}-ar"
+  ar = "${toolchain_bin}/llvm-ar"
   ld = "${toolchain_bin}/clang++"
-  readelf = "${toolchain_bin}/${custom_target_triple}-readelf"
-  nm = "${toolchain_bin}/${custom_target_triple}-nm"
-  strip = "${toolchain_bin}/${custom_target_triple}-strip"
+  readelf = "${toolchain_bin}/llvm-readelf"
+  nm = "${toolchain_bin}/llvm-nm"
+  strip = "${toolchain_bin}/llvm-strip"
 
   target_triple_flags = "--target=${custom_target_triple}"
   sysroot_flags = "--sysroot ${custom_sysroot}"
-- 
2.39.2
